src/main/kotlin/dev/cbyrne/kdiscordipc/data/activity/Activity.kt

```diff
@@ -7,12 +7,26 @@ import kotlinx.serialization.SerialName
 import kotlinx.serialization.Serializable
 import kotlinx.serialization.builtins.IntArraySerializer
 import kotlinx.serialization.builtins.serializer
+import kotlinx.serialization.descriptors.PrimitiveKind
+import kotlinx.serialization.descriptors.PrimitiveSerialDescriptor
 import kotlinx.serialization.descriptors.SerialDescriptor
 import kotlinx.serialization.encoding.*
 
+object TextFieldSerializer : KSerializer<String> {
+    override val descriptor: SerialDescriptor = PrimitiveSerialDescriptor("TextField", PrimitiveKind.STRING)
+
+    override fun deserialize(decoder: Decoder): String = decoder.decodeString()
+
+    override fun serialize(encoder: Encoder, value: String) {
+        encoder.encodeString(value.padEnd(2, ' '))
```

Comment:
This is going to add "  " to the end of every single string... not what we want. This is also not a solution to the problem, instead, we should have some sort of schema validation system.

Reply:
The thing is, discord thinks of text fields containing 2 spaces (e.g. `"  "`) as empty and displays them accordingly. This `KSerializer` is only used for the `state`, `details`, `large_text` and `small_text` fields (which as of now are the only ones that caused problems), rather than *all* strings (see `@Serializable(with = ...)`).